src/components/ShowList.jsx

```diff
@@ -50,27 +73,32 @@ function ShowList() {
               aria-label="clear filter text"
               className="clear-button"
             >
-              <i class="fas fa-times"></i>
+              <i className="fas fa-times"></i>
             </button>
           )}
           <ul>
             {filter
               ? value.docs
-                  .map((doc) => ({ id: doc.id, ...doc.data() }))
                   .filter((item) =>
-                    item.name.toLowerCase().includes(filter.toLowerCase()),
+                    item
+                      .data()
+                      .name.toLowerCase()
+                      .includes(filter.toLowerCase()),
                   )
+                  .sort(sortItems)
                   .map((item) => (
                     <Item
                       key={item.id}
-                      id={doc.id}
-                      lastPurchasedDate={item.lastPurchasedDate}
-                      name={item.name}
+                      id={item.id}
+                      lastPurchasedDate={item.data().lastPurchasedDate}
+                      name={item.data().name}
                     />
                   ))
-              : value.docs.map((doc) => (
-                  <Item key={doc.id} {...doc.data()} id={doc.id} />
-                ))}
+              : value.docs
+                  .sort(sortItems)
```

Comment:
good use of the builtin sort. 👍 

Extra credit: While an item is checked, move it to the bottom of the list.